framework/source/class/qx/core/Object.js

```diff
@@ -230,15 +230,25 @@ qx.Class.define("qx.core.Object",
     /**
      * Load user defined data from the object
      *
-     * @param key {String} the key
+     * @param key {String?} the key
+     *   If undefined, pass the user a copy of the entire map
      * @return {Object} the user data
      */
     getUserData : function(key)
     {
+      var data;
       if (!this.__userData) {
         return null;
       }
-      var data = this.__userData[key];
+      if (typeof key !== 'undefined')
+      {
+        data = this.__userData[key];
+      }
+      else
+      {
+        // give the caller a copy to protect our internal copy
+        data = JSON.parse(JSON.stringify(this.__userData));
```

Comment:
What if the user has stored something here that can't be JSON stringified? I agree that a copy of the user data should be returned, but I see cases where it's not possible. I'm not sure of the right solution to this.

Reply:
maybe a second parameter could specify if a copy is to be provided or not ... and then the copying could not only happen on full export but also on a single value

Reply:
Parameter added in d3e4d0b